ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java

```diff
@@ -110,26 +109,28 @@ private void registerForCreateSource(final ConfiguredStatement<? extends CreateS
   }
 
   private void registerForCreateAs(final ConfiguredStatement<? extends CreateAsSelect> cas) {
-    final ServiceContext sandboxServiceContext = SandboxedServiceContext.create(serviceContext);
-    final ExecuteResult executeResult = executionContext
-        .createSandbox(sandboxServiceContext)
-        .execute(sandboxServiceContext, cas);
-
-    final PersistentQueryMetadata queryMetadata = (PersistentQueryMetadata) executeResult
-        .getQuery()
-        .orElseThrow(() -> new KsqlStatementException(
-            "Could not determine output schema for query due to error: "
-                + executeResult.getCommandResult(),
-            cas.getStatementText()
-        ));
+    final CreateSourceCommand createSourceCommand;
+
+    try {
+      final ServiceContext sandboxServiceContext = SandboxedServiceContext.create(serviceContext);
+      createSourceCommand = (CreateSourceCommand)
+          executionContext.createSandbox(sandboxServiceContext)
+              .plan(sandboxServiceContext, cas)
+              .getDdlCommand()
+              .get();
+    } catch (final Exception e) {
+      throw new KsqlStatementException(
+          "Could not determine output schema for query due to error: "
+              + e.getMessage(), cas.getStatementText(), e);
+    }
 
     registerSchemas(
-        queryMetadata.getLogicalSchema(),
-        queryMetadata.getResultTopic().getKafkaTopicName(),
-        queryMetadata.getResultTopic().getKeyFormat().getFormatInfo(),
-        queryMetadata.getPhysicalSchema().keySchema().features(),
-        queryMetadata.getResultTopic().getValueFormat().getFormatInfo(),
-        queryMetadata.getPhysicalSchema().valueSchema().features(),
+        createSourceCommand.getSchema(),
+        createSourceCommand.getTopicName(),
+        createSourceCommand.getFormats().getKeyFormat(),
+        createSourceCommand.getFormats().getKeyFeatures(),
+        createSourceCommand.getFormats().getValueFormat(),
+        createSourceCommand.getFormats().getValueFeatures(),
```

Comment:
I had to build the query information from the plan instead of executing it. This avoids failing with an error if the `IF NOT EXISTS` keyword is used in the CREATE_AS statement. Also, this seems a better approach than execution 'cause this injector is only registering the schema in SR.